src/api/app/models/product.rb

```diff
@@ -21,7 +21,8 @@ def self.find_by_name_and_package( name, package )
   def self.all_products( project, expand = nil )
     return project.expand_all_products if expand
 
-    joins(:package).where("packages.project_id = ? and packages.name = '_product'", project.id)
+    joins("INNER JOIN packages ON packages.id = products.package_id
+           INNER JOIN package_kinds ON packages.id = package_kinds.package_id").where("packages.project_id = ? AND package_kinds.kind = 'product'", project.id)
```

Comment:
and how would I do that there with a nested one? .joins(..).joins(..) and neither joins(..,..) did not work.